packages/hop-node/src/watchers/BondTransferRootWatcher.ts

```
@@ -232,6 +257,85 @@ class BondTransferRootWatcher extends BaseWatcher {
+    // Validate logs with backup RPC endpoint, if it exists
+    const calculatedDbTransferRoot = await this.getCalculatedDbTransferRoot(txParams)
+    const blockNumber = calculatedDbTransferRoot.commitTxBlockNumber
+
+    const redundantRpcUrls = getRedundantRpcUrls(this.chainSlug) || []
+    for (const redundantRpcUrl of redundantRpcUrls) {
+      const redundantProvider = new providers.JsonRpcProvider(redundantRpcUrl)
+
+      // TODO: Better way to do this
+      const l2Bridge = contracts.get(this.tokenSymbol, this.chainSlug)?.l2Bridge
+      const events = await l2Bridge.connect(redundantProvider).queryFilter(
+        l2Bridge.filters.TransfersCommitted(),
+        blockNumber,
+        blockNumber
+      )
+      const eventParams = events.filter((x: any) => x.args.rootHash === txParams.transferRootHash)[0]
+      if (!eventParams) {
+        throw new PreTransactionValidationError(`TransferSent event not found for transferRootHash ${txParams.transferRootHash} at block ${blockNumber}`)
+      }
+
+      // TODO: better way to do this
+      if (
+        (Number(eventParams.args.destinationChainId) !== txParams.destinationChainId) ||
+        (eventParams.args.rootHash !== txParams.transferRootHash) ||
+        (eventParams.args.totalAmount.toString() !== txParams.totalAmount.toString()) ||
+        (eventParams.args.rootCommittedAt.toString() !== txParams.rootCommittedAt.toString())
+      ) {
+        throw new PreTransactionValidationError(`TransfersCommitted event does not match db. eventParams: ${JSON.stringify(eventParams)}, calculatedDbTransfer: ${JSON.stringify(calculatedDbTransferRoot)}`)
+      }
+    }
+  }
+
+  async getCalculatedDbTransferRoot (txParams: SendBondTransferRootTxParams): Promise<TransferRoot> {
+    const { transferRootHash, totalAmount } = txParams
+    const calculatedTransferRootId = getTransferRootId(transferRootHash, totalAmount)
+    return this.db.transferRoots.getByTransferRootId(calculatedTransferRootId)
```

suggestion: maybe throw `PreTransactionValidationError` error if db item is not found